### DISCUSSION TITLE
Improved Window Management

### DISCUSSION THREAD

**Discussion by T1mVo** [mrkai77/Loop#575] [Q&A]
While creating #574 I got quite confused by the window target calculation. Why do we calculate the window actions for for the preview and the target window? Wouldnt it be easier to only calculate the preview window, save it position and size and then just set the target window equal to the preview window?

**Comment by mrkai77** [comment]
Sure, that's possible. I guess it's currently like that for two reasons:
1. The previous window preview system did *not* use an actual window, and it instead used an overlay with a SwiftUI view contained within. This meant that it wouldn't be possible to apply the pixels directly from `WindowAction.getFrame()`, especially on multi-monitor setups. So that architecture carried onto the current preview system, even though it behaves almost exactly like a normal window.
1. The preview can be disabled. That said, this can easily be worked around, so not a huge reason :)

**Comment by mrkai77** [comment]
Moved to #700
